tests/utils/node/node.go

```diff
@@ -0,0 +1,275 @@
+// Copyright 2022 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package node
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os/exec"
+	"testing"
+
+	"github.com/ChainSafe/gossamer/dot/config/toml"
+	"github.com/ChainSafe/gossamer/lib/utils"
+	"github.com/ChainSafe/gossamer/tests/utils/config"
+	"github.com/ChainSafe/gossamer/tests/utils/pathfinder"
+	"github.com/stretchr/testify/require"
+)
+
+// Node is a structure holding all the settings to
+// configure a Gossamer node.
+type Node struct {
+	index      *int
+	configPath string
+	tomlConfig toml.Config
+	writer     io.Writer
+	logsBuffer *bytes.Buffer
+	binPath    string
+}
+
+// New returns a node configured using the
+// toml configuration and options given.
+func New(t *testing.T, tomlConfig toml.Config,
+	options ...Option) (node Node) {
+	node.tomlConfig = tomlConfig
+	for _, option := range options {
+		option(&node)
+	}
+	node.setDefaults(t)
+	node.setWriterPrefix()
+	node.configPath = config.Write(t, node.tomlConfig)
+	return node
+}
+
+func (n Node) String() string {
+	indexString := fmt.Sprint(*n.index)
+	return fmt.Sprintf("%s-%s", n.tomlConfig.Account.Key, indexString)
+}
+
+// GetRPCPort returns the rpc port of the node.
+func (n Node) GetRPCPort() (port string) { return fmt.Sprint(n.tomlConfig.RPC.Port) }
+
+// GetWSPort returns the websocket port of the node.
+func (n Node) GetWSPort() (port string) { return fmt.Sprint(n.tomlConfig.RPC.WSPort) }
+
+// GetKey returns the key of the node.
+func (n Node) GetKey() (key string) { return n.tomlConfig.Account.Key }
```

Comment:
nit: remove the `Get`

Reply:
I'll add it to the engineering wiki Go page